cmd/git-sync/main.go

```diff
@@ -695,6 +707,18 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 
 	// Make a worktree for this exact git hash.
 	worktreePath := filepath.Join(gitRoot, hash)
+
+	// Avoid wedge cases where the worktree was created but this function error'd without cleaning the worktree.
+	// Next timearound, the sync loop fails to create the worktree and bails out.
+	// Error observed:  
+	//   " Run(git worktree add /repo/root/rev-nnnn origin/develop):
+	//     exit status 128: { stdout: \"Preparing worktree (detached HEAD nnnn)\\n\", stderr: \"fatal: '/repo/root/rev-nnnn' already exists\\n\" }"  
+	//.  "
+	if err := cleanupWorkTree(ctx, gitRoot, worktreePath); err != nil {
```

Comment:
Would the clean up fail before the worktree is created? Should we only call clean up after we fail to add the worktree?

Reply:
os.RemoveAll() returns nil if path is not present.
git prune is a maintanence command and not parameterized to the worktree.